diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt

```diff
@@ -0,0 +1,62 @@
+package org.cqfn.diktat.ruleset.rules.files
+
+import com.pinterest.ktlint.core.KtLint
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.BLOCK
+import com.pinterest.ktlint.core.ast.ElementType.CLASS_BODY
+import com.pinterest.ktlint.core.ast.ElementType.LBRACE
+import com.pinterest.ktlint.core.ast.ElementType.RBRACE
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.TOO_MANY_BLANK_LINES
+import org.cqfn.diktat.ruleset.rules.getDiktatConfigRules
+import org.cqfn.diktat.ruleset.utils.leaveExactlyNumNewLines
+import org.cqfn.diktat.ruleset.utils.leaveOnlyOneNewLine
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+class BlankLinesRule : Rule("blank-lines") {
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
```

Comment:
this visitor should check and fix that code blocks are separated by new lines, please create a feature for this. 
We don't want to have such code:
```
package 
import
fun foo() {
}
fun foo1() {
}
```

we need to fix such code

Reply:
if you think that should be in a separate rule - let's add such rule to a code style

Reply:
I believe it's [ClassLikeStructuresOrderRule](https://github.com/cqfn/diKTat/blob/master/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/ClassLikeStructuresOrderRule.kt) that already does this - it rearranges code blocks and separates them with newlines.

Reply:
It's [Rule 3.2](https://github.com/cqfn/diKTat/blob/master/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/ClassLikeStructuresOrderRule.kt) of code style.